agent/pkg/builder/container/container_builder.go

```diff
@@ -400,41 +402,54 @@ func prepareImage(dc *DockerContainerBuilder) error {
 	return nil
 }
 
-func createNetwork(dc *DockerContainerBuilder) *string {
-	networkName := fmt.Sprintf("%s-network", dc.networkMode)
+func createNetworks(dc *DockerContainerBuilder) []string {
+	var networkIds []string
 
-	filter := filters.NewArgs()
-	filter.Add("name", networkName)
+	for _, network := range dc.networks {
```

Comment:
The `network` variable could be called `networkName` so the extra variable could be eliminated.